monkey/agent_plugins/exploiters/snmp/src/community_string_generator.py

```diff
@@ -10,15 +10,19 @@
     Username,
 )
 from infection_monkey.exploit.tools import identity_type_filter, secret_type_filter
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
 
 
-def generate_community_strings(credentials: Iterable[Credentials]) -> Iterable[str]:
+def generate_community_strings(
+    propagation_credentials_repository: IPropagationCredentialsRepository,
+) -> Iterable[str]:
```

Comment:
There's no reason to couple this function to `IPropagationCredentialsRepository`. We can modify the code that calls this to pass in credential and then also drop the changes to the unit tests.